design/local-node-agent.md

```diff
@@ -91,17 +107,72 @@ The experience is also normalized across other distributed storage platforms tha
 Instead of the user having to create a storageclass specific for the underlying distributed storage (e.g., RBD if Rook is using Ceph), this will abstract that decision away, allowing the user to simply request storage.
 This was already true of the hybrid approach in Rook v0.5, but it's worth noting here still.
 
+#### Volume Attach CRD Details
+This section will discuss the volume attach CRD in more detail.
+The name (primary identity) of each CRD instance will be the PVC name that it is for.
+This will enable very efficient look ups of volume attach instances for a particular PVC, thereby making fencing checks very efficient.
+
+Each CRD instance will track all consumers of the volume.
+In the case of `ReadWriteOnce` there will only be a single consumer, but for `ReadWriteMany` there can be multiple simultaneous consumers.
+There will be an `Attachments` list that captures each instance of pod, node, and mount dir, which can all be used for the fencing checks described in the next section.
+
+The full schema of the volume attachment CRD is shown below:
+```go
+type Volumeattachment struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata"`
+    Attachments       []Attachment `json:"attachments"`
+}
+
+type Attachment struct {
+    Node         string `json:"node"`
+    PodNamespace string `json:"podNamespace"`
+    Pod          string `json:"pod"`
+    MountDir     string `json:"mountDir,omitempty"`
+}
+```
+
 #### Fencing
-Ensuring exclusive single client access (`ReadWriteOnce`) to the persistent volume can be achieved by the Rook agent, independently from the underlying storage protocol that is being used (RBD, NBD, iSCSI, etc.).
+Ensuring either exclusive single client access (`ReadWriteOnce`) or shared multi-pod access (`ReadWriteMany`) to the persistent volume can be achieved by the Rook agent,
+independently from the underlying storage protocol that is being used (RBD, NBD, iSCSI, etc.).
 Remember that during the attach operation, the Rook agent will create a volume attach CRD.
-When the agent performs the attach on its local node, it can update the CRD with information stating that the volume is now locked.
+When the agent performs the attach on its local node, it can update the CRD with information stating that the volume is now locked and who it is locked by.
 Thus, the CRD itself is providing the accounting for the volume's lock in a generalized way, independent of the of the underlying storage protocol the agent chose to use for the volume.
 Of course, this lock can be augmented at the lower layer of the specific storage protocol, e.g., `rbd lock add` for RBD.
```

Comment:
should we remove this section of `rbd lock` since we are not using it?

Reply:
i removed the `rbd lock add` example, but i'm keeping the general notion that the CRD lock can be augmented at lower layers if desired.